src/parser.c

```diff
@@ -776,10 +776,7 @@ enum parse_rv parse_ImportTransaction(struct ImportTransactionState *const state
       switch (state->state) {
         case 0: // ChainID
             CALL_SUBPARSER(id32State, Id32);
-            if(is_pchain_transaction(meta->type_id)) {
-              if(memcmp(network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->id32State.buf, sizeof(blockchain_id_t)))
-                REJECT("Invalid XChain ID");
-            } else {
+            if(!is_pchain_transaction(meta->type_id)) {
```

Comment:
What does `is_pchain_transaction` mean, in the context of cross-chain transactions?

Reply:
That is allows a few different things makes me nervous, I think I will want to inline the case to be more precise.

Reply:
This is the code that is called in the "`is-not-pchain-transaction`" branch of the `if` statement that existed before the PR. I'm down to improve the check to be more precise, but what should the case be if not this?

Reply:
[cross-annotating off-GitHub conversation to here: John is investigating this and might come back with more questions]